schema/types/acname_test.go

```diff
@@ -40,3 +40,22 @@ func TestNewACNameBad(t *testing.T) {
 		}
 	}
 }
+
+func TestSanitizeACName(t *testing.T) {
+	tests := map[string]string{
+		"foo#":                                             "foo-",
```

Comment:
I have a conflicting proposed change in #185, namely that a valid ACName should not begin or end with namespace or separation characters (the last three characters in ACNameValChars). Could this be modified to check for and handle those cases?